pkg/apps/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -160,6 +160,15 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 
 		return c.updateStatus(config, existingDeployments)
 	}
+
+	// Never deploy with invalid or unresolved images
+	for i, container := range config.Spec.Template.Spec.Containers {
+		if strings.TrimSpace(container.Image) == "" {
```

Comment:
i thought we guard against this in `shouldTrigger()`

Reply:
we don't, we wait for the RC to be created and because the image resolution/injection is happening asynchronously it will manage to deploy RC with "" before the image gets injected

Reply:
I tried fixing the image that function but then it will end up being triggered as config change. we could probably add exception for that as well but this felt cleanest and most reliable. We should do the same upstream.

Reply:
would it be better to move this up to `triggerActivated` and just return false, true ?

Reply:
I guess it could be there now that shouldSkip is honored

Reply:
@mfojtik actually that wouldn't work for manual rollouts without any trigger, so need to stay here